app/models/institutions_archive.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  def self.archive_previous_versions
+    production_version = Version.current_production.number
+    previous_version = Version.previous_production.number
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(previous_version, production_version)
+        Institution.where('version >= ? and version < ?', previous_version, production_version).delete_all
```

Comment:
Is will this ever delete anything other than `previous_version`?
```suggestion
       `Institution.where(version: previous_version).delete_all`
```

Reply:
There can be multiple preview versions between production versions